### PR TITLE
add dns server for node syncing

### PR DESCRIPTION
**Pull request by chaosma**
* add dns server for node syncing
* add dns flag in main to turn on dns server syncing
* don't print viewID checking when in syncing mode

### CODE REVIEW

File: api/service/syncing/syncing.go [hunk 1]
```diff
@@ -637,6 +637,8 @@ func (ss *StateSync) getMaxPeerHeight() uint64 {
 		wg.Add(1)
 		go func() {
 			defer wg.Done()
+			//debug
+			//utils.GetLogInstance().Warn("[Sync] getMaxPeerHeight", "IP", peerConfig.ip, "Port", peerConfig.port)
```
Comment (LeoHChen) [inline api/service/syncing/syncing.go]: please remove the debug code or use Debug()

File: cmd/harmony/main.go [hunk 2]
```diff
@@ -84,6 +84,8 @@ var (
 	metricsReportURL = flag.String("metrics_report_url", "", "If set, reports metrics to this URL.")
 	versionFlag      = flag.Bool("version", false, "Output version info")
 	onlyLogTps       = flag.Bool("only_log_tps", false, "Only log TPS if true")
+
+	dnsFlag = flag.Bool("dns", false, "use harmony dns nodes if true; use libp2p peer discovery if false")
```
Comment (LeoHChen) [inline cmd/harmony/main.go]: not aligned?

File: node/node.go [hunk 3]
```diff
@@ -505,3 +506,9 @@ func (node *Node) initNodeConfiguration() (service.NodeConfig, chan p2p.Peer) {
 func (node *Node) AccountManager() *accounts.Manager {
 	return node.accountManager
 }
+
+// SetDNSFlag indicates whether use harmony dns server to get peer info for node syncing
+func (node *Node) SetDNSFlag(flag bool) {
+	utils.GetLogInstance().Info("SetDNSFlag", "flag", flag)
```
Comment (LeoHChen) [inline node/node.go]: This is a debug message, not an info as there is nothing more informative.

File: node/node_syncing.go [hunk 4]
```diff
@@ -28,6 +29,11 @@ const (
 	MinConnectedPeers = 5  // minimum number of peers connected to in node syncing
 )
 
+var (
+	// DNS harmony dns server for harmony nodes IP
+	DNS = [4]string{"s0.t.hmny.io", "s1.t.hmny.io", "s2.t.hmny.io", "s3.t.hmny.io"}
```
Comment (LeoHChen) [inline node/node_syncing.go]: why don't you put it in a constant? and add into some internal config package?  like internal/configs/net?  this configuration can be used by other packages like wallet.

File: node/node_syncing.go [hunk 5]
```diff
@@ -65,6 +71,25 @@ func (node *Node) GetSyncingPeers() []p2p.Peer {
...
+	if err != nil {
+		utils.GetLogInstance().Debug("GetPeersFromDNS cannot find peers", "error", err)
+		return nil
+	}
+	port := syncing.GetSyncingPort(node.SelfPeer.Port)
+	peers := []p2p.Peer{}
+	for _, addr := range addrs {
+		peers = append(peers, p2p.Peer{IP: addr, Port: port})
+	}
+	//utils.GetLogInstance().Debug("dns server returns:", "addrs", addrs)
```
Comment (LeoHChen) [inline node/node_syncing.go]: remove the comments or the line.

File: api/service/syncing/syncing.go (hunk 1 above)
Comment (chaosma) [reply api/service/syncing/syncing.go]: I might use it later for debug

File: cmd/harmony/main.go (hunk 2 above)
Comment (chaosma) [reply cmd/harmony/main.go]: what do you mean aligned?

File: node/node_syncing.go (hunk 5 above)
Comment (chaosma) [reply node/node_syncing.go]: sure

File: node/node_syncing.go (hunk 4 above)
Comment (chaosma) [reply node/node_syncing.go]: sure

Comment (chaosma) [reply node/node_syncing.go]: by the way, the string array cannot be a constant, so I use var

File: node/node.go (hunk 3 above)
Comment (chaosma) [reply node/node.go]: yes, but it can be helpful to debug whether the command line is passed correctly

**Pull request merged**